lib/datasets/dataset.rb

```diff
@@ -33,9 +33,23 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, url, *fallback_urls, &block)
+      urls = [url] + fallback_urls
+      urls.each_with_index do |url, idx|
```

Comment:
`i` or `index` is better than `idx`:

```suggestion
      urls.each_with_index do |url, index|
```

Reply:
I agree, using `i` or `index` is clearer than `idx`.